classes/plugins/PKPPubIdPluginHelper.inc.php

```diff
@@ -195,11 +195,9 @@ function clearPubId($contextId, $pubIdPlugInClassName, $pubObject) {
 				if (get_class($pubIdPlugin) == $pubIdPlugInClassName) {
 					// clear the pubId:
 					// delete the pubId from the DB
-					$pubObjectType = $pubIdPlugin->getPubObjectType($pubObject);
-					$daos = $pubIdPlugin->getDAOs();
-					$dao = $daos[$pubObjectType];
+					$dao = $pubObject->getDAO();
 					$pubObjectId = $pubObject->getId();
-					if ($pubObjectType == 'SubmissionFile') {
+					if (is_a($pubObject, 'SubmissionFile')) {
```

Comment:
I think this exception was necessary because we had some confusing inheritance cases -- see the note in `SubmissionFile.inc.php`'s `getFileId` function:

    // WARNING: Do not modernize getter/setters without considering
    // ID clash with subclasses ArticleGalley and ArticleNote!

I think those cases have both been removed, so we can move to using `getId`/`setId` on `SubmissionFile` subclasses instead of `getFileId`/`setFileId`. I think that means this special case can be removed and `getFileId`/`setFileId` can be removed from `SubmissionFile`, letting calls there fall back to `PKPFile`, which uses the more standard `getId` and `setId`.